src/hotspot/share/opto/inlinetypenode.cpp

```diff
@@ -460,69 +471,31 @@ void InlineTypeNode::store_flattened(GraphKit* kit, Node* base, Node* ptr, ciIns
 
 void InlineTypeNode::store(GraphKit* kit, Node* base, Node* ptr, ciInstanceKlass* holder, int holder_offset, DecoratorSet decorators) const {
   // Write field values to memory
+  int field_idx = 0;
   for (uint i = 0; i < field_count(); ++i) {
-    int offset = holder_offset + field_offset(i);
     Node* value = field_value(i);
     ciType* ft = field_type(i);
+    int offset = holder_offset + field_offset(i);
     if (field_is_flattened(i)) {
-      if (kit->gvn().type(value)->isa_vect()) {
-        const TypePtr* adr_type = field_adr_type(base, offset, holder, decorators, kit->gvn());
-        Node* adr = kit->basic_plus_adr(base, ptr, offset);
-        int vec_len = kit->gvn().type(value)->is_vect()->length();
-        BasicType elem_bt = kit->gvn().type(value)->is_vect()->element_basic_type();
-        bool bundle_size_supported = Matcher::vector_size_supported(elem_bt, vec_len);
-
-        // Set the vector length to  maximal supported vector length to allow
-        // graceful compilation exit at a later stage.
-        vec_len = bundle_size_supported ? vec_len : Matcher::max_vector_size(elem_bt);
-
-        Node* store = kit->gvn().transform(StoreVectorNode::make(0, kit->control(), kit->memory(adr), adr, adr_type, value, vec_len));
-        kit->set_memory(store, adr_type);
-
-        if (!bundle_size_supported) {
-          kit->env()->record_method_not_compilable("Mutifield bundle size not supported for target", false);
-        }
-      } else {
-        if (!value->is_InlineType()) {
-          // Recursively store the flattened inline type field
-          value = InlineTypeNode::make_from_oop(kit, value, ft->as_inline_klass());
-        }
-        value->as_InlineType()->store_flattened(kit, base, ptr, holder, offset, decorators);
+      if (!value->is_InlineType()) {
+        // Recursively store the flattened inline type field
+        value = InlineTypeNode::make_from_oop(kit, value, ft->as_inline_klass());
       }
+      value->as_InlineType()->store_flattened(kit, base, ptr, holder, offset, decorators);
     } else {
-      // Store field value to memory
-      const TypePtr* adr_type = field_adr_type(base, offset, holder, decorators, kit->gvn());
-      Node* adr = kit->basic_plus_adr(base, ptr, offset);
+      int vec_len = ft->bundle_size();
       BasicType bt = type2field[ft->basic_type()];
-      assert(is_java_primitive(bt) || adr->bottom_type()->is_ptr_to_narrowoop() == UseCompressedOops, "inconsistent");
       const Type* val_type = Type::get_const_type(ft);
       const TypeAryPtr* ary_type = kit->gvn().type(base)->isa_aryptr();
       if (ary_type != NULL) {
         decorators |= IS_ARRAY;
       }
```

Comment:
These code could be removed to the `else` branch in line-499.

Reply:
Done